src/webgpu/util/conversion.ts

```diff
@@ -498,19 +498,26 @@ export class Scalar {
       return Colors.bold(this.value.toString());
     }
     switch (this.value) {
-      case 0:
       case Infinity:
       case -Infinity:
         return Colors.bold(this.value.toString());
       default: {
+        // Uint8Array.map returns a Uint8Array, so cannot use .map directly
+        const hex = Array.from(this.bits)
+          .reverse()
+          .map(x => x.toString(16).padStart(2, '0'))
+          .join('');
         const n = this.value as Number;
         if (n !== null) {
-          return (
-            Colors.bold(this.value.toString()) +
-            `(0x${this.value.toString(16)}, subnormal: ${isSubnormalNumber(n.valueOf())})`
-          );
+          const str =
+            isFloatValue(this) && Number.isInteger(n)
```

Comment:
nit: `1.0e+20` is an integer, but shouldn't be printed as `1.0e+20.0`.

For #1515, I used the helper:

```
    const valueWithPoint = (x: number) => {
      const str = `${x}`;
      return str.indexOf('.') > 0 ? str : `${str}.0`;
    };
```

Reply:
I agree that my version is not correct either, but I think your version still adds in unexpected `.0`s

For `1e+21` I am seeing `1e+21.0`

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZjAvDAjABg5r2cG4BQokIANgKYB0JIA5gBQAGAJAN5wC+DAlAUdDADcAhiQCuZAOoBLKAAsACiClhYyOgA8AXDDCiAtgCMyAJy5IAfDBb4YtmH1jRjSGMxbrOBOzGNkoo4zAYJwplABMydQB5ODoAcgo4s0s0GAB+YKhnbTcndgo0BgJ2XnAIUkpqejdhMUkZBSUVOjguTh4gA

Reply:
Humm, good spot. I thought it always printed the `.0` when it's going to emit the `e`. This seems to work:

```
const valueWithPoint = (x: number) => {
    const str = `${x}`;
    return (str.indexOf('.') > 0) || (str.indexOf('e') > 0) ? str : `${str}.0`;
};
```

Reply:
That looks like it catches all of the cases I can quickly think of :+1: